pkg/argocd/applications.go

```diff
@@ -202,3 +202,23 @@ func (builder *ApplicationBuilder) validate() (bool, error) {
 
 	return true, nil
 }
+
+// WithGitDetails applies git details to application definition.
+func (builder *ApplicationBuilder) WithGitDetails(gitRepo, gitBranch, gitPath string) *ApplicationBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
```

Comment:
Please add argument verification. 
You need to make sure that arguments are not empty strings

Reply:
Done, thanks.